### PR TITLE
Issue #2846 Jaas ldap unit test

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

File: jetty-jaas/src/test/java/org/eclipse/jetty/jaas/JAASLdapLoginServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+        userIdentity = ls.login( "someoneelse", "verycomplicatedpassword", request);
+        assertNotNull( userIdentity );
+        assertFalse( userIdentity.isUserInRole( "developers", null) );
+        assertTrue( userIdentity.isUserInRole( "admin", null) );
+        assertFalse( userIdentity.isUserInRole( "blabla", null) );
+    }
+
+    @Test
+    public void testLdapUserIdentityBindingLogin() throws Exception
```
Comment (janbartel) [inline jetty-jaas/src/test/java/org/eclipse/jetty/jaas/JAASLdapLoginServiceTest.java]: These new tests look very nice!

File: jetty-jaas/src/main/java/org/eclipse/jetty/jaas/spi/LdapLoginModule.java [hunk 2]
```diff
@@ -556,7 +556,7 @@ private SearchResult findUser(String username) throws NamingException, LoginExce
             throw new LoginException("User not found.");
         }
 
-        return (SearchResult)results.nextElement();
+        return results.nextElement();
     }
```
[the remark this answers is not in the document]
Comment (olamy) [reply jetty-jaas/src/main/java/org/eclipse/jetty/jaas/spi/LdapLoginModule.java]: done

**Comment by janbartel** [comment]
@olamy love it. I will merge it now.

File: jetty-jaas/src/test/java/org/eclipse/jetty/jaas/JAASLdapLoginServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+/**
+ * JAASLdapLoginServiceTest
+ *
+ *
+ */
+@RunWith( FrameworkRunner.class)
```
Comment (joakime) [inline jetty-jaas/src/test/java/org/eclipse/jetty/jaas/JAASLdapLoginServiceTest.java]: Uh....   Is this going to be another junit4 stapled test like our osgi paxexam?

**Pull request merged**